### DISCUSSION TITLE
Gatsby build on Netlify failing with the objects must have internal.contentDigest

### DISCUSSION THREAD

**Discussion by viddyoze** [gatsbyjs/gatsby#37694] [Help]
Hi. I have a Gatsby/Wordpress site running on Netlify and the build will fail unless I manually run the build and clear cache option. This is the error I'm receiving 

```
error the objects must have internal.contentDigest. Current object:
{
  "objectID": "cG9zdDozNDc=",
  "uri": "/post-name/",
  "title": "Post Name",
  "excerpt": "This is a post..."
}
```
I'm not sure what to try, any help is appreciated

**Comment by 3200pro** [comment]
@viddyoze  I'm having the same problem with a Gatsby site running on Gatsby Cloud with Sanity.io content. I'll reach out through my Gatsby Cloud account and hopefully get a quick answer from them.

**Reply by aidanm1999** [reply]
@3200pro Check out my answer below if you are using `graphql` with your Sanity CMS. If you are using `groq`, I would request the document's `id` and pass that into the transformer:

```typescript
export const transformer = (items) => items.map((item) => ({
    objectID: post.id,
    [...],
    internal: {
        contentDigest: post.id,
    }
});
```

> Note: I haven't tested this myself so *may* still have the issue

**Comment by aidanm1999** [comment]
Hi @viddyoze,

I believe the issue is related to `gatsby-plugin-algolia` during the `transformer` phase of the build. To fix you should add in `internal { contentDigest }` to your query, e.g.:

```graphql
allWpPost {
    nodes {
        id,
        [...],
        internal { contentDigest }
    }
}
``` 

and then during the `transformer` function, when formatting your array, you should add the internal object in:

```typescript
export const transformer = (data) => data?.allWpPost?.nodes.map((post) => ({
    objectID: post.id,
    [...],
    internal: post.internal,
});
```

That should sort your builds out. NB if you are not using the `transformer` function, you should be able to just add `internal { contentDigest }` to your `graphql` query and that should fix the issue.
